Check benefits update March 2023

Rosa-Fox opened the pull request:
[Trello](https://trello.com/c/IdKAZVYV/352-check-benefits-and-financial-support-smart-answer)

Implements the changes that are [highlighted in Pink on this Spreadsheet](https://docs.google.com/spreadsheets/d/1-qpD5JHsl7fzkY0tmMdt-3Z3S97yZhiXms_QlWt_nC8/edit?usp=sharing). Let me know if you need access to it. There is also further info in the comments on the spreadsheet.

Changes:

- Are you in paid work? (question 3) has different options and hint text (hint text is line 12 of the spreadsheet). Users will be navigated to question 3a on answering 'yes'.
- New question 'How many paid hours work do you do?' (question 3a).
- Update hint text for question 4a.
- New question 'Does your disability or health condition affect your ability to do daily tasks or get around?' (question 4b). Users are show this on answering 'yes' to 'Do you have a disability or health condition?' (question 4a).
- Updated disability_affecting_work question (4c) title to 'Does your disability or health condition affect your ability to work or mean you need support to work?' and new option added.

These changes meant that the existing logic for lots of the outcomes in the calculator needed to be updated. I tackled the calculator method changes and their corresponding tests in individual commits. I put the updates to flow tests for the questions in one larger commit as I found it difficult to separate it out. I also made a separate commit for fixing the existing outcome flow tests.

There were then some new outcomes to add (line 84-90 in the spreadsheet). 

There were various content changes. These can be viewed by looking at the comments on cells in:

Line 23 - Warm Home Discount Scheme result text and move down in results (England and Wales)
Line 24 - Warm Home Discount Scheme result text and move down in results (Scotland)
Line 28 - Access to Work result text
LIne 29 - Access to Work result link (NI)
Line 32 - Scottish Child Payment result text
Line 43 - Sure Start Maternity Grant result text
Line 57 - Education Maintenance Allowance result text
Line  70 - Wales fuel support scheme payment result text
Lines 86 and 87 - Help to Save logic
[Generic result page](https://docs.google.com/document/d/1g3BSZUM1Q5_u00CcM-GtJmVR4-7brqy-DZWpmuLcJ5c/edit?pli=1#heading=h.bo4c7mcmliy1) - scroll down doc for one comment


⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

gclssvglx commented on test/unit/calculators/checks_benefits_financial_support_calculator_test.rb:
```diff
@@ -1097,7 +1097,75 @@ class CheckBenefitsFinancialSupportCalculatorTest < ActiveSupport::TestCase
+          should "be false if child not living with you, child has a disability and less than 16000 in assets and savings" do
+            @calculator.children_living_with_you = "no"
+            @calculator.children_with_disability = "yes"
+            %w[none_16000 under_16000].each do |asset|
+              @calculator.assets_and_savings = asset
+              assert_not @calculator.eligible_for_help_with_house_adaptions_if_you_are_disabled?
+            end
```
Same logic as adult maybe?

Rosa-Fox replied:
👍 have fixed

gclssvglx commented on lib/smart_answer/calculators/check_benefits_financial_support_calculator.rb:
```diff
@@ -295,6 +295,9 @@ def eligible_for_help_to_save?
       disabled_adult_less_than_16000_savings? || living_with_disabled_child_less_than_16000_savings?
     end
+
+    def eligible_for_help_with_house_adaptions_if_you_are_disabled?
+      disabled_adult_less_than_16000_savings? || living_with_disabled_child_less_than_16000_savings?
```
I wonder if this differs in Scotland...?

gclssvglx reviewed (commented):
Wow! Amazing work @Rosa-Fox - I've no idea how you kept track of all these complex changes 🥇 

I'm sure most of my comments are due to my confusion, but would be good to talk through some of these.

Rosa-Fox commented on config/smart_answers/check_benefits_financial_support_data.yml:
```diff
@@ -636,10 +636,22 @@
+      description: <p>You could get a grant from your local council to make changes to your home if you’re disabled or you live with someone who is.</p>
+      condition: eligible_disabled_facilities_grant?
+      url_text: Check if you’re eligible for a Disabled Facilities Grant
+      url: https://www.gov.uk/disabled-facilities-grants
```
ah so that is in the following chunk of YAML, they use the same method but different content from the YAML file. I updated the name of the following bit of YAML to be called `- name: disabled_facilities_grant_ni` to make it a bit clearer and avoid a name clash!

gclssvglx replied:
Gotcha!

Rosa-Fox commented on config/smart_answers/check_benefits_financial_support_data.yml:
```diff
@@ -636,10 +636,22 @@
         - northern-ireland
+    - name: disabled_facilities_grant
+      title: Disabled Facilities Grant
+      description: <p>You could get a grant from your local council to make changes to your home if you’re disabled or you live with someone who is.</p>
```
same as above, the NI text in the next bit of YAML

gclssvglx replied:
No worries, I did the original review by commit, so didn't see this. Will use file view this time 😉

Rosa-Fox commented on test/unit/calculators/checks_benefits_financial_support_calculator_test.rb:
```diff
@@ -1097,7 +1097,75 @@ class CheckBenefitsFinancialSupportCalculatorTest < ActiveSupport::TestCase
 
+      context "eligible_for_help_with_house_adaptions_if_you_are_disabled? for a disabled adult" do
+        context "when eligible" do
+          should "be true with a disability or health condition and less than 16000 in assets and savings" do
+            @calculator.disability_or_health_condition = "yes"
+            %w[none_16000 under_16000].each do |asset|
+              @calculator.assets_and_savings = asset
+              assert @calculator.eligible_for_help_with_house_adaptions_if_you_are_disabled?
+            end
+          end
```
ooh my mistake! I'll fix this tomorrow!

Rosa-Fox commented on test/unit/calculators/checks_benefits_financial_support_calculator_test.rb:
```diff
@@ -1097,7 +1097,75 @@ class CheckBenefitsFinancialSupportCalculatorTest < ActiveSupport::TestCase
 
+            @calculator.assets_and_savings = "over_16000"
+            assert_not @calculator.eligible_for_help_with_house_adaptions_if_you_are_disabled?
+          end
```
ooh my mistake! I'll fix this tomorrow!

Rosa-Fox commented on test/unit/calculators/checks_benefits_financial_support_calculator_test.rb:
```diff
@@ -1097,7 +1097,75 @@ class CheckBenefitsFinancialSupportCalculatorTest < ActiveSupport::TestCase
+            @calculator.children_with_disability = "yes"
+
+            @calculator.assets_and_savings = "over_16000"
+            assert_not @calculator.eligible_for_help_with_house_adaptions_if_you_are_disabled?
```
👍 have fixed

Rosa-Fox commented on test/flows/check_benefits_financial_support_flow_test.rb:
```diff
@@ -347,44 +348,56 @@ class CheckBenefitsFinancialSupportFlowTest < ActiveSupport::TestCase
                           over_state_pension_age: "no",
-                          are_you_working: work_hours,
                           disability_or_health_condition: "yes",
+                          disability_affecting_daily_tasks:,
```
yeah :(

Rosa-Fox commented on config/smart_answers/check_benefits_financial_support_data.yml:
```diff
@@ -653,5 +653,11 @@
+      condition: eligible_disabled_facilities_grant?
+      url_text: Check if you’re eligible for a Disabled Facilities Grant on the NI Housing Executive website
+      url: https://www.nihe.gov.uk/housing-help/grants/types-of-grants-available/disabled-facilities-grant
+      countries:
```
sorry I didn't make it easy to review commit by commit!

gclssvglx replied:
No worries 😄

Rosa-Fox commented:
> disabled_facilities_grant

I renamed the NI version of `disabled_facilities_grant` to `disabled_facilities_grant_ni` in the YAML file. Thanks for noticing the logic differed for 'How much money, savings and investments do you have (excluding the home you live in)?' between the the 'Help with house adaptations if you are disabled' and 'Disabled facilities Grant'. I have fixed this up. It is ready for re-review! I'll send the preview link to the content designers to they can start reviewing too.

gclssvglx reviewed (approved):
Great work @Rosa-Fox - a very complex and confusing set of changes. Brilliantly navigated 🏅
